src/components/Grid/GridUtils.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { Card } from '../Card';
+import { OverlappingCard } from '../OverlappingCard';
+import { StripCard } from '../StripCard';
+import {
+  DEFAULT_MIN_WIDTH,
+  CARD_MIN_WIDTH,
+  OVERLAPPING_CARD_MIN_WIDTH,
+  STACKED_CARD_MIN_WIDTH,
+  STRIP_CARD_MIN_WIDTH,
+} from './constants';
+
+export function itemsPerRowWidth(rowWidth, itemWidth, maxColumns, columnGap) {
+  if (rowWidth !== 0) {
+    while (rowWidth < maxColumns * itemWidth + columnGap * (maxColumns - 1)) {
+      maxColumns--;
+    }
+  }
+
+  return maxColumns || 1;
+}
+
+export function getMediaQueries({
+  maxColumns,
+  minColumnWidth,
+  columnGap,
+  ListItemClass,
+  gridId,
+}) {
+  let mediaQueries = '';
+  while (maxColumns > 0) {
+    const minWidth = maxColumns * minColumnWidth + columnGap * (maxColumns - 1);
+    mediaQueries = `@media (min-width: ${minWidth}px) {#${gridId} .${ListItemClass} {-ms-grid-columns: repeat(${maxColumns}, minmax(${minColumnWidth}px, 100vw));grid-template-columns: repeat(${maxColumns}, minmax(${minColumnWidth}px, 100vw));}}${mediaQueries}`;
+    maxColumns--;
+  }
+  return <style dangerouslySetInnerHTML={{ __html: mediaQueries }} />;
+}
+
+export function getMinWidthByCardType(CardComponent): number {
```

Comment:
Is this really necessary? 
I think it adds some "noise" to this component. 
You will obviously know which items you're passing to this component, so you can use the `minColumnWidth` prop.
And other users might want to use cards with different min width, so it seems very specific.

Reply:
we removed the min-width from the cards. we still need to enforce the the min-width.
is exporting the min-width from each card and ask users to enforce it in the MD is better solution?

Reply:
But this is logic that can be inside of your app. I'm not sure this should be part of the component.
The component can accept a `minWidth` prop, and have the default `DEFAULT_MIN_WIDTH`.

Reply:
removed

Reply:
We cannot remove it, there is minimum width per card defined in the design system,
A grid cannot "take" the minimum from a cell (and maintain all the other logic that we need), so if the min for the card is higher than 1fr it will just overlap on the next grid column, which is not what we want, so this is a must,
considering that the api allows to set it by the tpa, we can have the logic in Bookings, but we already discussed it with DS team, and we agreed that the grid can support non-card elements, but optimized for them, so we do need this logic in the library.